src/bosh_aws_cpi/spec/unit/resize_disk_spec.rb

```diff
@@ -0,0 +1,83 @@
+require "spec_helper"
+
+describe Bosh::AwsCloud::CloudV1, "resize_disk" do
+  let(:volume_manager) { instance_double(Bosh::AwsCloud::VolumeManager) }
+  let(:volume) { instance_double(Aws::EC2::Volume, :id => 'disk-id') }
+  let(:volume_resp) { instance_double(Aws::EC2::Types::ModifyVolumeResult, :volume_modification => {modification_state: 'completed'}) }
+
+  before do
+    @cloud = mock_cloud do |_ec2|
+      @ec2 = _ec2
+      allow(@ec2).to receive(:config).and_return('fake-config')
+      allow(@ec2).to receive(:volume).with("disk-id").and_return(volume)
+    end
+    allow(volume).to receive(:size).and_return(2)
+    allow(volume).to receive(:attachments).and_return([])
+    allow(@ec2.client).to receive(:modify_volume).and_return(volume_resp)
+    modification = instance_double(Bosh::AwsCloud::SdkHelpers::VolumeModification, :state => 'completed')
+    allow(Bosh::AwsCloud::SdkHelpers::VolumeModification).to receive(:new).with(volume, volume_resp.volume_modification, @ec2.client).and_return(modification)
+    allow(Bosh::AwsCloud::ResourceWait).to receive(:for_volume_modification).with(volume_modification: modification, state: 'completed')
+  end
+
+
+  it 'uses the AWS endpoint to resize a disk' do
+    allow(Bosh::Clouds::Config.logger).to receive(:info)
+    return_value = @cloud.resize_disk('disk-id', 4096)
+
+    expect(return_value).to eq(nil)
+
+    expect(@ec2.client).to have_received(:modify_volume).with(volume_id: 'disk-id',size: 4)
+    expect(Bosh::Clouds::Config.logger).to have_received(:info).with('Disk disk-id resized from 2 GiB to 4 GiB')
+  end
+
+  context 'when trying to resize to the same disk size' do
+    it 'does not call extend on disk and writes to the log' do
+      allow(Bosh::Clouds::Config.logger).to receive(:info)
+
+      @cloud.resize_disk('disk-id', 2048)
+
+      expect(Bosh::Clouds::Config.logger).to have_received(:info).with('Skipping resize of disk disk-id because current value 2 GiB is equal new value 2 GiB')
+    end
+  end
+
+  context 'when trying to resize disk to a new size with an not even size in MiB' do
+    it 'does not call extend on disk and writes to the log' do
```

Comment:
Why should it not call extend when resizing a disk with an not even size?
Also the test at line 50 actually tests that the method is called.

Reply:
@rkoster  Sorry for the late reply. I just mimicked with my tests the behaviour of the openstack CLI: https://github.com/cloudfoundry/bosh-openstack-cpi-release/blob/master/src/bosh_openstack_cpi/spec/unit/resize_disk_spec.rb#L44
I'm ok with removing this, I thought this was the expected behaviour.

Reply:
@beyhan Do you have any context on where that test in the openstack cpi came from? It seems like the test does not match its description. The test explicitly tests for extend to be called: https://github.com/cloudfoundry/bosh-openstack-cpi-release/blob/master/src/bosh_openstack_cpi/spec/unit/resize_disk_spec.rb#L54

Reply:
@rkoster The error in the description seems to be a copy&paste failure from the previous test (https://github.com/cloudfoundry/bosh-openstack-cpi-release/blob/a8fa225aa75db1df3c40d62c843b00b68ba385e4/src/bosh_openstack_cpi/spec/unit/resize_disk_spec.rb#L33). I corrected the test description and also extended it to fit to the reality. The PR is: https://github.com/cloudfoundry/bosh-openstack-cpi-release/pull/240

Reply:
@rkoster  @beyhan  Thanks for clarifying. I now renamed the description similar to cloudfoundry/bosh-openstack-cpi-release#240